nav2_costmap_2d/plugins/static_layer.cpp

```diff
@@ -72,12 +72,14 @@ StaticLayer::onInitialize()
 
   getParameters();
 
-  rclcpp::QoS map_qos(1);
+  rclcpp::QoS map_qos(10); // initialize to default
   if (map_subscribe_transient_local_) {
     map_qos.transient_local();
+    map_qos.reliable();
```

Comment:
you could use map_quos.transient_local().reliable().keep_last(1)

Reply:
In my opinion, that's harder to read. Maybe I'm old.